lib/Sema/CSStep.h

```diff
@@ -1033,7 +1031,6 @@ class ConjunctionStep : public BindingStep<ConjunctionElementProducer> {
 private:
   /// Restore best and current scores as they were before conjunction.
   void restoreCurrentScore(const Score &solutionScore) const {
-    CS.CurrentScore = CurrentScore;
```

Comment:
I think this method needs to be renamed now since it no longer restores the score, only adds hole + fixes scores from the closure to indicate that the body failed to type-check.

Reply:
We need to do this to make sure that we rank solutions with invalid closures appropriately and don’t produce a valid solution if a multi-statement closure failed.